### ISSUE TITLE
Boost 1.67 breaks the homebrew build

### ISSUE THREAD

**Issue by phlptp** [GMLC-TDC/HELICS-src#266]
Boost 1.67 breaks the homebrew build.  Boost 1.67 came out at the same time as the 1.0.0 release,  looks like there will need to be a couple fixes to restore compatibility.    We will have to make sure the fixes work for both the new boost and previous versions.

**Comment by wangli1030** [comment]
Hi,
I just tried to install HELICS in Windows and Mac. I tried several ways to install it. But I always end up with this error.

> [ 13%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpComms.cpp.o
/Users/belle/Documents/GitHub/HELICS-src/src/helics/core/tcp/TcpComms.cpp:140:15: error: 
      invalid operands to binary expression ('const boost::system::error_code'
      and 'nullptr_t')
    if (error != nullptr)
        ~~~~~ ^  ~~~~~~~
1 error generated.
make[2]: *** [src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpComms.cpp.o] Error 1
make[1]: *** [src/helics/core/CMakeFiles/helics_core.dir/all] Error 2
make: *** [all] Error 2

I think new version of Boost might be the reason. Therefore, I posted it in here.
Thank you.

**Comment by kdheepak** [comment]
@wangli1030 Can you try again with the latest master?

**Comment by wangli1030** [comment]
Thank you.
@kdheepak It is working now for both Windows and Mac. 
I have questions about Test section in the instruction. I did not find exact directory or folder shown in both Windows and Mac instruction.  For Windows, there is no 'player' folder under 'helics'.
For Mac, there is no folder called 'helics_install'. Does it mean the installation is unsuccessful? I did not see any errors during both installation.
Thank you.
